specification.md

```diff
@@ -1,15 +1,279 @@
 # Specification
 
+This document serves as the 0.1 specification for SBOMit (SBOM on in-toto).
+The specification should not be considered stable at this time.
+
 ## Introduction
 
+### Motivation
+A major problem with the state of an SBOM today is that the documents are 
+often inaccurate.  This is largely because the documents are derived by looking
+at the resulting software and trying to understand what happened in the past.
+
+This specification proposes a means to generate metadata for an SBOM while the
+software is being created.  Furthermore, the means by which this information
+is captured uses (in-toto)[https://in-toto.io] attestations and layouts.  This 
+provides cryptograpic validation that this information is correct, handles 
+key distribution and management to indicate which parties should be trusted
+for each step, and captures information about the environment in which the 
+steps are run.  
+
+As a result, using SBOMit provides a more accurate SBOM when parties are 
+honest.  When malicious parties interfere in the process, SBOMit provides 
+a mix of traceability (knowing which party was malicious) and prevention 
+(blocking malicious software from being trusted), depending on how the 
+in-toto steps are configured..
+
+
+### Definitions
+
+SIT -- An SBOM which is derived from a SBOMit document.  It can
+be in any SBOM format.  It points back to the SBOMit document, which was used
+to generate it. 
+
+SBOMit document -- A long form document which contains sufficient information
+to verify the supply chain process is correct.  It may be used to generate one
+or more SIT.
+
 ## Threat model
 
-### Root of trust for layouts
+One key goal of our design is to provide as much security as possible in all 
+cases.  Security should degrade gracefully when the attacker gains new 
+capabilities.  Hence, we will consider attacker models where the attacker is 
+exceedingly powerful and try to restrict the damage they can do even in those 
```

Comment:
"exceedingly powerful" is still a little bit wishy-washy. Can they, for example, compromise all of the keys used to sign root layouts?

Reply:
Yes.  I list that case below.